feat: Add a high-level page about the dev toolbar (#7503)
withastro/docs: src/content/docs/en/guides/dev-toolbar.mdx

:::note
The performance and accessibility audits done by the dev toolbar are intentionally not as comprehensive as those provided by dedicated tools like [Pa11y](https://pa11y.org/) or [Lighthouse](https://developers.google.com/web/tools/lighthouse), or even better, humans.

The dev toolbar aims to provide a quick and easy way to catch common issues during development, without needing to context-switch to a different tool.
:::

### Settings

The Settings app allows you to toggle different settings for the development toolbar, such as verbose logging, and the ability to disable notifications.

## Extending the dev toolbar

[Astro integrations](/en/guides/integrations-guide/) can add new apps to the dev toolbar, allowing you to extend it with custom tools that are specific to your project. To see a list of integrations that provide dev toolbar apps, see the "Dev Toolbar" section of the [integrations list](https://astro.build/integrations/?search=&categories%5B%5D=toolbar).

## Disabling the dev toolbar

### Per-project

> To disable the dev toolbar for everyone working on a project, you can set `devToolbar: false` in your project's [config file](/en/reference/config-reference/).

Reviewer: Found a broken link here

```suggestion
To disable the dev toolbar for everyone working on a project, you can set `devToolbar: false` in your project's [config file](/en/reference/configuration-reference/).
```

---
title: Dev Toolbar
---

Astro includes a development toolbar at the bottom of every page. This toolbar includes a number of useful tools ("apps") for debugging and inspecting your site during development, and can be extended using [Astro integrations](/en/guides/integrations-guide/) with the [Dev Toolbar API](/en/reference/dev-toolbar-app-reference/).

This toolbar is enabled by default and will only appear in development mode. It is not included in the final build of your site.

## Built-in apps

### Astro

The Astro app provides easy access to different information about the current project and links about Astro. Notably, it provides a one-click access to the Astro documentation, GitHub repository, and Discord server.

This app also include a "Copy debug info" button, which will copy the output of the [`astro info`](/en/reference/cli-reference/#astro-info) command to your clipboard. This can be useful when asking for help or reporting issues.

### X-ray

> The X-ray app allows you to inspect the [Islands](/en/concepts/islands/) on the current page, showing you the properties that are being passed to each islands, and the client directive that is being used to render them.

Reviewer: ```suggestion
The X-ray app allows you to inspect the [Islands](/en/concepts/islands/) on the current page, showing you the properties that are being passed to each island, and the client directive that is being used to render them.
```

:::note
The performance and accessibility audits done by the dev toolbar are intentionally not as comprehensive as those provided by dedicated tools like [Pa11y](https://pa11y.org/) or [Lighthouse](https://developers.google.com/web/tools/lighthouse), or even better, humans.

The dev toolbar aims to provide a quick and easy way to catch common issues during development, without needing to context-switch to a different tool.
:::

### Settings

The Settings app allows you to toggle different settings for the development toolbar, such as verbose logging, and the ability to disable notifications.

## Extending the dev toolbar

[Astro integrations](/en/guides/integrations-guide/) can add new apps to the dev toolbar, allowing you to extend it with custom tools that are specific to your project. To see a list of integrations that provide dev toolbar apps, see the "Dev Toolbar" section of the [integrations list](https://astro.build/integrations/?search=&categories%5B%5D=toolbar).

> ```shell

Reviewer: Is this accidentally added? 👀

## Disabling the dev toolbar

### Per-project

To disable the dev toolbar for everyone working on a project, you can set `devToolbar: false` in your project's [config file](/en/reference/config-reference/).

```js title="astro.config.mjs"
import { defineConfig } from "astro/config";

export default defineConfig({
  devToolbar: {
    enabled: false
  }
})
```

### Per-user

If you want to disable the dev toolbar for yourself on a specific project, you can do so using the `astro preferences` command.

```shell
astro preferences disable devToolbar
```

> To disable the devToolbar in all Astro projects on the current machine:

Reviewer: nit: for word consistency

```suggestion
To disable the dev toolbar in all Astro projects on the current machine:
```

```shell
astro preferences disable --global devToolbar
```

> The devToolbar can later be enabled with:

Reviewer: nit: for word consistency

```suggestion
The dev toolbar can later be enabled with:
```
